packages/components/src/visually-hidden/index.js

```diff
@@ -0,0 +1,14 @@
+
+function VisuallyHidden( {
+	children,
+	isInline = false,
+} ) {
+	const TagName = isInline ? 'span' : 'div';
```

Comment:
Reakit uses an interesting pattern, they have `as` prop which you can pass to change the tag name, or even use a component. See: https://reakit.io/docs/composition/#as-prop

```js
<VisuallyHidden as="span">
	This should not show.
</VisuallyHidden>
```

They also offer an alternative way to use render props. See: https://reakit.io/docs/composition/#render-props

```js
< VisuallyHidden>
      { (  { children } ) => (
        <span {...props}>{ children }</span> 
      ) }
</VisuallyHidden>
```

@diegohaz, how much work would it be to replicate that here?

Reply:
`as` seem to "web" focused for me especially if we want the API of these components to be DOM agnostic. and the render props are great but I'd like to know if there's a potential use-case that the current API doesn't solve?

Reply:
In Reakit `as` can be a string or Component, the native version:

```js
<VisuallyHidden as={ View }>
    This should not show.
</VisuallyHidden>
```

Reply:
mmm interesting

Reply:
It's simple to implement:
```jsx
function renderAsRenderProps({ as: T = "div", ...props }) {
  if (typeof props.children === "function") {
    return props.children(props);
  }
  return <T {...props} />;
}

function Component({ as = "span", ...props }) {
  return renderAsRenderProps({ as, ...props });
}
```

Using children as a function is better than the `as` prop to compose multiple components:
```jsx
<Component>
  {props => <div><Button {...props} /></div>}
</Component>
```

Reply:
The `as` prop is quite popular nowadays. The first library I saw using it was [Semantic UI](https://react.semantic-ui.com/#augmentation). Not sure if there are other libraries combining it with render props though. But I think it's a simple and powerful combination.

Reply:
Thanks @diegohaz - I updated the component to use the "as" method as described, works well.

I defined the `renderAsRenderProps` function within the VisuallyHidden component, not exported. It  seems useful that we may want to use it elsewhere. If so, where would I move it to?